File: java/kata-03-wrap-method/src/test/java/tech/qmates/kata/hr/EmployeeServiceTest.java
package tech.qmates.kata.hr;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

/**
 * Skeleton test class — no @Test methods yet; that is your job.
 *
 * TODO: Drive the NEW publishing method test-first, in isolation. Don't call
 *       promote(...) (it hits the database) — test the method directly: inject
 *       an ObservableEventBus and assert on bus.publishedEvents().
 *       `new EmployeeService()` is cheap, so no mocks or subclassing needed.
 *
 *       Grow the newLeadership rule one test at a time: it is true when newTitle
 *       starts (case-insensitively) with "Head", "Chief" or "Manager".
 *         - "Head of Sales"   -> newLeadership true
 *         - "Senior Engineer" -> newLeadership false
 *         - "chief of staff"  -> newLeadership true (case-insensitive)
 *
 * // @Test
 * // void setsLeadershipFlagForLeadershipTitle() {
 * //     ObservableEventBus bus = new ObservableEventBus();
 * //     new EmployeeService().publishPromotionEvent(bus, "E-1", "Head of Sales");
 * //     // assert one PromotionEvent whose newTitle is "Head of Sales" and newLeadership is true
 * // }
 */
class EmployeeServiceTest {
}

File: typescript/kata-03-wrap-method/tests/employeeService.test.ts
import { describe, it } from 'vitest';

describe('promotion event publishing (wrapped method)', () => {
  it.todo('sets newLeadership = true for leadership titles (Head/Chief/Manager)');
  it.todo('sets newLeadership = false for non-leadership titles');
  it.todo('matches leadership titles case-insensitively');
  // Drive the NEW publishing method test-first, in isolation. Don't call
  // promote(...) (it hits the database) — test the method directly: inject an
  // ObservableEventBus and assert on bus.publishedEvents(). `new EmployeeService()`
  // is cheap, so no mocks or subclassing needed.
  //
  // The rule: newLeadership is true when newTitle starts (case-insensitively)
  // with "Head", "Chief" or "Manager", and false otherwise.
  //
  // import { EmployeeService } from '../src/employeeService';
  // import { ObservableEventBus } from './observableEventBus';
  //
  // it('sets newLeadership = true for a leadership title', () => {
  //   const bus = new ObservableEventBus();
  //   new EmployeeService().publishPromotionEvent(bus, 'E-1', 'Head of Sales');
  //   // assert one PromotionEvent with toTitle 'Head of Sales' and newLeadership true
  // });
});
